common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java

```diff
@@ -117,43 +127,99 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
             // Always draw a black background to cover transparent map areas
             RenderUtils.drawRect(poseStack, CommonColors.BLACK, renderX, renderY, 0, width, height);
 
-            List<MapTexture> maps = MapRenderer.getMapTextures(
-                    (int) McUtils.player().getX(), (int) McUtils.player().getZ(), width, height, scale);
+            // enable rotation if necessary
+            if (followPlayerRotation) {
+                poseStack.pushPose();
+                RenderUtils.rotatePose(
+                        poseStack, centerX, centerZ, 180 - McUtils.player().getYRot());
+            }
+
+            // avoid rotational overpass - This is a rather loose oversizing, if possible later
+            // use trignometry, etc. to find a better one
+            float extraFactor = 1f;
+            if (followPlayerRotation) {
+                // 1.5 > sqrt(2);
+                extraFactor = 1.5F;
+
+                if (width > height) {
+                    extraFactor *= width / height;
+                } else {
+                    extraFactor *= height / width;
+                }
+            }
+
+            List<MapTexture> maps = MapModel.getMapsForBoundingBox(textureBoundingBox);
             for (MapTexture map : maps) {
-                float textureX = map.getTextureXPosition(McUtils.player().getX());
-                float textureZ = map.getTextureZPosition(McUtils.player().getZ());
+                float textureX = map.getTextureXPosition(playerX);
+                float textureZ = map.getTextureZPosition(playerZ);
                 MapRenderer.renderMapQuad(
                         map,
                         poseStack,
-                        (float) McUtils.player().getX(),
-                        (float) McUtils.player().getZ(),
                         centerX,
                         centerZ,
                         textureX,
                         textureZ,
-                        width,
-                        height,
+                        width * extraFactor,
+                        height * extraFactor,
                         this.scale,
-                        this.poiScale,
-                        null,
-                        false,
-                        this.followPlayerRotation,
                         this.renderUsingLinear);
             }
 
-            // TODO minimap icons
+            // disable rotation if necessary
+            if (followPlayerRotation) {
+                poseStack.popPose();
+            }
+
+            List<Poi> pois = MapModel.getAllPois()
```

Comment:
Do you think we should have a `MapModel.getPoisForBoundingBox`, to do some initial filtering?

Reply:
I don't think that such a method is necessary, as we can filter in `MainMapScreen` when calculating hovered and `MinimapFeature` has rotation, which makes filtering also annoying.

Though I agree some easy, more general checks could be good, I'm unable to think of how we could do one